### ISSUE TITLE
Add comprehensive training pipeline documentation

### ISSUE THREAD

**Issue by jbarnes850** [Arc-Computer/atlas-sdk#124]
## Problem Statement

The Atlas SDK README and documentation mention the runtime → training closed loop as a core value proposition, but there is no dedicated documentation explaining how to set up, configure, and execute the training pipeline. Staff+ ML Engineers evaluating Atlas need clear guidance on:

1. How to set up Atlas Core alongside the SDK
2. What the export format looks like and what data is required
3. How to configure Hydra parameters for GRPO/GKD/SFT training
4. How to load trained checkpoints back into SDK agents
5. What "good training" looks like (metrics, convergence, iteration workflow)

This gap creates significant friction for users trying to complete the full continual learning loop.

## Current State

**What exists:**
- `atlas train` CLI command with comprehensive flags (discovered via `--help`)
- README.md mentions training integration (lines 200-225)
- Brief Python API example for `get_training_sessions()` (README.md:173-182)
- Reference to Atlas Core repository

**What's missing:**
- Dedicated training pipeline guide in docs/
- Atlas Core setup and prerequisites documentation
- JSONL export schema documentation
- Hydra configuration reference for available trainer/data/model configs
- Checkpoint deployment guide (trained model → production agent)
- Training metrics interpretation and debugging guidance
- A/B testing framework documentation
- Iterative improvement workflow (collect → train → deploy → repeat)

## Proposed Solution

Create `docs/training/training_pipeline.md` with the following sections:

### 1. Prerequisites and Setup
- Atlas Core installation and dependencies
- Environment variable configuration (ATLAS_CORE_PATH, STORAGE__DATABASE_URL)
- Python version compatibility between SDK and Core
- Dependency conflicts and resolution

### 2. Data Collection Phase
- Minimum number of sessions required for training
- Task variation best practices for generalization
- Reward quality validation before training
- Review workflow (approve/quarantine sessions)

### 3. Export and Format
- JSONL schema documentation with annotated examples
- `AtlasSessionTrace` structure breakdown
- Direct Postgres access vs JSONL export trade-offs
- Export filtering strategies (min_reward, review_status, learning_key)

### 4. Training Configuration
- Hydra config structure overview
- Available trainer configs (grpo, gkd, base_sft, teacher_grpo)
- When to use each training method (speed/quality trade-offs)
- Common override parameters with examples:
  - `trainer.max_steps`
  - `trainer.learning_rate`
  - `data.batch_size`
  - `model.gradient_checkpointing`
- WandB integration setup

### 5. Launching Training
- `atlas train` command examples with real use cases
- `--dry-run` workflow for validation
- `--use-sample-dataset` for testing wiring
- Monitoring training progress (stdout, WandB, checkpoints)
- Error troubleshooting guide

### 6. Checkpoint Deployment
- Where checkpoints are saved (outputs/ directory structure)
- Checkpoint format (HuggingFace, safetensors)
- Loading checkpoints into SDK agents
- Configuration changes needed for fine-tuned models

### 7. Production Workflow
- A/B testing baseline vs fine-tuned agents
- Measuring improvement in production
- Iterative refinement loop
- Monitoring continual learning in production

### 8. Training Recipes
Provide copy-paste examples for common scenarios:

```bash
# Recipe 1: Quick GRPO training on approved sessions
export STORAGE__DATABASE_URL=postgresql://atlas:atlas@localhost:5433/atlas
atlas train \
  --config-name offline/base \
  --trainer-config grpo \
  --wandb-project my-agent-training \
  --override trainer.max_steps=100 \
  --limit 50

# Recipe 2: Fast GKD distillation (9-30x faster than GRPO)
atlas train \
  --config-name offline/base \
  --trainer-config gkd \
  --wandb-project my-agent-training \
  --limit 100

# Recipe 3: SFT on high-quality traces only
atlas train \
  --config-name offline/base \
  --trainer-config base_sft \
  --include-status approved \
  --override data.min_reward=0.8 \
  --limit 200
```

## Integration Points

This documentation should cross-reference:
- `docs/operations/guardrails.md` - Review workflow
- `docs/evaluation/learning_eval.md` - Learning metrics
- `docs/evaluation/reward_eval.md` - Judge validation
- `examples/mcp_tool_learning/README.md` - Working example
- Atlas Core repository README

## Success Criteria

After this documentation is added, a Staff ML Engineer should be able to:

- Set up the training pipeline in under 30 minutes
- Understand which training method to use for their scenario
- Launch a training run without reading Atlas Core source code
- Deploy a trained checkpoint back into their SDK agent
- Iterate on the training loop independently

## Related Issues

This is part of the broader onboarding audit (Phase 4: End-to-End Training Pipeline Audit).

## Additional Context

The `atlas train` command is well-implemented with comprehensive CLI flags, but the feature is undiscoverable without dedicated documentation. The README mentions it briefly, but users need a full guide to understand the end-to-end workflow.

**Comment by jbarnes850** [comment]
## Assessment

After investigating this issue, I've reviewed the Atlas Core documentation and can confirm that **comprehensive training pipeline documentation already exists**:

| Documentation | Location | Coverage |
|--------------|----------|----------|
| Step-by-step training guide | [GRPO Training Guide](https://docs.arc.computer/training/offline/grpo-training) | Complete 3-step pipeline (SFT → vLLM → GRPO), troubleshooting, validation |
| Data export & access | [Export Runtime Traces](https://docs.arc.computer/sdk/export-traces) | Direct DB access + JSONL export methods |
| Training data pipeline | [Training Data Pipeline](https://docs.arc.computer/training/offline/training-data-pipeline) | API reference, filtering, pagination, performance optimization |
| Hydra configuration | [Training Configuration](https://docs.arc.computer/training/configuration) | Exhaustive parameter reference for all trainers |
| GKD training | [GKD Training Guide](https://docs.arc.computer/training/offline/gkd-training) | 9-30x faster distillation method |

**Root cause**: This is a **discoverability issue**, not a missing documentation issue. The Core docs are comprehensive but SDK users don't know they exist.

---

## Solution: SDK Documentation Updates

Add targeted signposting in the SDK repository to guide users to existing Core training documentation:

### 1. SDK README - Add "Training Pipeline" section

After the runtime quickstart (around line 100-150), add:

\`\`\`markdown
## Training Your Model

Once you've collected runtime traces, use **Atlas Core** to train updated teacher models:

1. **Export your data**: \`arc-atlas --database-url postgresql://... --output traces.jsonl\`
2. **Train with Core**: See [Atlas Core Training Guide](https://docs.arc.computer/training/offline/grpo-training)
3. **Deploy checkpoint**: Point SDK at your new model in runtime config

For direct database access (faster, no exports needed): [Training Data Pipeline](https://docs.arc.computer/training/offline/training-data-pipeline)

**Training Methods**:
- **GRPO** - Reinforcement learning from reward signals ([Guide](https://docs.arc.computer/training/offline/grpo-training))
- **GKD** - 9-30x faster distillation for production models ([Guide](https://docs.arc.computer/training/offline/gkd-training))
- **SFT** - Supervised fine-tuning on approved traces

See the [Training Configuration Guide](https://docs.arc.computer/training/configuration) for Hydra parameters and advanced options.
\`\`\`

### 2. SDK docs/operations/ - Add training callouts

In export/review documentation, add a "Next Steps" section:

\`\`\`markdown
## Next Steps: Training

Runtime traces feed the offline training pipeline in Atlas Core:

- [GRPO Training Guide](https://docs.arc.computer/training/offline/grpo-training) - Complete step-by-step pipeline
- [Training Configuration](https://docs.arc.computer/training/configuration) - Hydra parameters reference
- [GKD Training](https://docs.arc.computer/training/offline/gkd-training) - Fast distillation (9-30x faster than GRPO)
- [Training Data Pipeline](https://docs.arc.computer/training/offline/training-data-pipeline) - Direct database access API

**Common workflow**:
1. Collect traces with SDK runtime
2. Review/approve sessions (\`arc-atlas review\`)
3. Train updated model in Core (GRPO/GKD/SFT)
4. Deploy checkpoint back to SDK agents
5. Repeat
\`\`\`

### 3. SDK docs/getting-started/ - Add training overview

Add a brief "Learning Loop" or "Training Integration" page that explains:
- The SDK collects data, Core trains models
- Link to Core training guides
- How to load trained checkpoints back into SDK agents
- A/B testing baseline vs fine-tuned agents

---

## Why This Approach

- **Minimal changes**: 2-3 SDK documentation additions
- **Leverages existing content**: Core docs are already excellent and comprehensive
- **Clear separation of concerns**: SDK = runtime orchestration, Core = offline training
- **Follows user journey**: SDK users collect traces → discover training options → navigate to Core docs
- **Avoids duplication**: No need to duplicate Core training content in SDK

---

## Atlas Core Changes

The Core repository documentation is already comprehensive. The only potential addition would be a dedicated "Checkpoint Deployment" section in the training guide that explicitly covers:
- Loading trained checkpoints into SDK runtime config
- Verifying checkpoint compatibility
- A/B testing trained vs baseline models

But this may be better suited for SDK documentation since it's SDK-specific configuration.

---

## Next Steps

If this approach looks good:
1. I can draft the specific SDK documentation additions as PRs
2. Add any missing "checkpoint deployment" details to Core docs if needed
3. Close this issue once SDK documentation is updated

Let me know if you'd like me to proceed with the SDK documentation PRs or if you prefer a different approach.

**Comment by jbarnes850** [comment]
## Update: PR #131 Also Resolves Issue #126

PR #131 has been updated to address both this issue (#124) and issue #126 (HuggingFace model path documentation).

The PR now includes:

**For #124 (Training Pipeline Documentation):**
- Links to complete GRPO/GKD/SFT training guides
- Training data pipeline documentation
- Hydra configuration reference
- "Next Steps: Training" sections in guardrails and export docs

**For #126 (Checkpoint Deployment):**
- Concrete config examples for HuggingFace Inference Endpoints
- Local inference server configuration (vLLM/TGI)
- Proper `api_base` and `api_key_env` usage
- Continual learning loop explanation

Both issues can be closed when PR #131 merges: https://github.com/Arc-Computer/atlas-sdk/pull/131
